mongo-core/src/main/java/io/micronaut/configuration/mongo/core/serde/SerdeCodec.java

```diff
@@ -73,7 +74,7 @@ class SerdeCodec<T> implements Codec<T> {
     @Override
     public T decode(BsonReader reader, DecoderContext decoderContext) {
         try {
-            return deserializer.deserialize(new BsonReaderDecoder(reader), this.decoderContext, argument);
+            return deserializer.deserialize(new BsonReaderDecoder(reader, LimitingStream.DEFAULT_LIMITS), this.decoderContext, argument);
```

Comment:
@sdelamo Compile was failing without this. The change is introduced in https://github.com/micronaut-projects/micronaut-serialization/pull/496
Maybe they should left default constructor and pass default like this. Not sure if this is correct fix, but there is no config for this currently in mongodb. Maybe @yawkat can advise what is the correct approach here

Reply:
Now test is failing because platform includes M12 and this change is in M13